app/models/miq_request_workflow.rb

```diff
@@ -1042,7 +1042,10 @@ def ems_folder_to_hash_struct(ci)
   end
 
   def storage_to_hash_struct(ci)
-    build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type])
+    storage_cluster = ci.storage_clusters.detect { |cluster| cluster.ems_id == get_source_and_targets[:ems].try(:id) }
```

Comment:
@bdunne see my comment here https://github.com/ManageIQ/manageiq/pull/7874#discussion_r59398714
If you have a datastore shared across datacenters VMware prevents you from adding it to a storage cluster, and in a datacenter a datastore can only be in one storage cluster